io/src/csv/read_cells.cpp

```diff
@@ -148,10 +146,20 @@ void read_cells(cell_collection_types::host& cells, std::string_view filename,
             ddIndex = it->second;
         }
 
+        // Helper lambda for setting up SoA cells.
+        auto assign = [ddIndex](auto soa_cell, const cell& iocell) {
+            soa_cell.channel0() = iocell.channel0;
+            soa_cell.channel1() = iocell.channel1;
+            soa_cell.activation() = iocell.value;
+            soa_cell.time() = iocell.timestamp;
+            soa_cell.module_index() = ddIndex;
+        };
```

Comment:
Did you investigate the possibility of adding an emplace function? Because this is still far from ideal, unfortunately. The main concern is that any change in the structure of the cell, esp. the addition of new fields, will go unnoticed by the compiler; this is a big bug surface.

Reply:
I'm open to a concrete code suggestion. Unfortunately I couldn't come up with a solution to this yet.

Note that with xAODs we've never been able to do it. Efficiently at least. Because with the xAOD code in order to do this, we have to create a temporary "auxiliary store" for the standalone object that we would then use to emplace back into the container. I.e. the object that we give to `emplace_back(...)` still has to point to some container. (At least with the xAOD infrastructure it does.)

It may not be impossible to make this all happen. :thinking: But the idea I have right now, will need a fair amount of development. (I'd need to create a whole new type based on `vecmem::edm::schema`, which would be the argument of `vecmem::edm::host<>::emplace_back(...)`.)

Something to keep in mind. But we can't stop this PR based on this, as it will never finish then. :thinking: